File: packages/android/src/main/java/com/kozmos/components/Pagination/Pagination.figma.kt
package com.kozmos.components.pagination

import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType
import com.figma.code.connect.FigmaVariant
import com.kozmos.tokens.KozmosColors

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=280-1157")
@FigmaVariant("Content", "Basic")
class KozmosPaginationBasicConnect {
    @FigmaProperty(FigmaType.Text, "Page 1 Text")
    val page1: String = "1"

    @FigmaProperty(FigmaType.Text, "Page 2 Text")
    val page2: String = "2"

    @FigmaProperty(FigmaType.Text, "Page 3 Text")
    val page3: String = "3"

    @Composable
    fun ComponentExample() {
        KozmosPagination {
            KozmosPaginationPrevious(onClick = {})
            KozmosPaginationLink(text = page1, onClick = {})
            KozmosPaginationLink(text = page2, isActive = true, onClick = {})
            KozmosPaginationLink(text = page3, onClick = {})
            KozmosPaginationNext(onClick = {})
        }
    }
}

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=280-1157")
@FigmaVariant("Content", "Ellipsis")
class KozmosPaginationEllipsisConnect {
    @FigmaProperty(FigmaType.Text, "Page 1 Text")
    val page1: String = "1"

    @FigmaProperty(FigmaType.Text, "Page 2 Text")
    val page2: String = "2"

    @FigmaProperty(FigmaType.Text, "Page 3 Text")
    val page3: String = "3"

    @FigmaProperty(FigmaType.Text, "Last Page Text")
    val lastPage: String = "10"

    @Composable
    fun ComponentExample() {
        KozmosPagination {
            KozmosPaginationPrevious(onClick = {})
            KozmosPaginationLink(text = page1, onClick = {})
            KozmosPaginationLink(text = page2, isActive = true, onClick = {})
            KozmosPaginationLink(text = page3, onClick = {})
            KozmosPaginationEllipsis()
            KozmosPaginationLink(text = lastPage, onClick = {})
            KozmosPaginationNext(onClick = {})
        }
    }
}

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=280-1157")
@FigmaVariant("Content", "Compact")
class KozmosPaginationCompactConnect {
    @FigmaProperty(FigmaType.Text, "Compact Text")
    val compactText: String = "2 / 10"

    @Composable
    fun ComponentExample() {
        KozmosPagination {
            KozmosPaginationPrevious(onClick = {})
            Text(
                text = compactText,
                color = KozmosColors.primitivesColorsForeground500,
                style = MaterialTheme.typography.bodyMedium
            )
            KozmosPaginationNext(onClick = {})
        }
    }
}

File: packages/ios/Sources/Components/Pagination/Pagination.figma.swift
import SwiftUI
import Figma

struct KozmosPaginationBasicConnect: FigmaConnect {
    let component = KozmosPagination<AnyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=280-1157"
    var variant = ["Content": "Basic"]

    @FigmaString("Page 1 Text")
    var page1: String = "1"

    @FigmaString("Page 2 Text")
    var page2: String = "2"

    @FigmaString("Page 3 Text")
    var page3: String = "3"

    var body: some View {
        KozmosPagination {
            KozmosPaginationPrevious(action: {})
            KozmosPaginationLink(text: self.page1, action: {})
            KozmosPaginationLink(text: self.page2, isActive: true, action: {})
            KozmosPaginationLink(text: self.page3, action: {})
            KozmosPaginationNext(action: {})
        }
    }
}

struct KozmosPaginationEllipsisConnect: FigmaConnect {
    let component = KozmosPagination<AnyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=280-1157"
    var variant = ["Content": "Ellipsis"]

    @FigmaString("Page 1 Text")
    var page1: String = "1"

    @FigmaString("Page 2 Text")
    var page2: String = "2"

    @FigmaString("Page 3 Text")
    var page3: String = "3"

    @FigmaString("Last Page Text")
    var lastPage: String = "10"

    var body: some View {
        KozmosPagination {
            KozmosPaginationPrevious(action: {})
            KozmosPaginationLink(text: self.page1, action: {})
            KozmosPaginationLink(text: self.page2, isActive: true, action: {})
            KozmosPaginationLink(text: self.page3, action: {})
            KozmosPaginationEllipsis()
            KozmosPaginationLink(text: self.lastPage, action: {})
            KozmosPaginationNext(action: {})
        }
    }
}

struct KozmosPaginationCompactConnect: FigmaConnect {
    let component = KozmosPagination<AnyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=280-1157"
    var variant = ["Content": "Compact"]

    @FigmaString("Compact Text")
    var compactText: String = "2 / 10"

    var body: some View {
        KozmosPagination {
            KozmosPaginationPrevious(action: {})
            Text(self.compactText)
                .font(.subheadline)
                .foregroundColor(KozmosColors.primitivesColorsForeground500)
            KozmosPaginationNext(action: {})
        }
    }
}
